File: api/services.py
import requests
from omeifeproj.settings import TOKEN

BASE_URL = "https://apis.omeife.ai/api/v1/"

def generate_api_key(TOKEN):
    url = BASE_URL + "user/developer/generate-key"

    headers = {
        'Authorization': f'Bearer {TOKEN}',
        'Content-Type': 'application/json'
    }

    response = requests.get(url, headers=headers)

    response_json = response.json() 

    """
    Return only API KEY
    """
    return response_json["data"]["key"]

def make_request(endpoint, data):
    KEY = generate_api_key(TOKEN)
    print(KEY)
    
    headers = {
        'Authorization': f'Bearer {KEY}',
        'Content-Type': 'application/json'
    }

    """
    Create endpoint
    """
    full_enpoint = BASE_URL + endpoint
    
    """
    Send requests
    """
    response = requests.post(full_enpoint, headers=headers, json=data)


    """
    Return response
    """
    return response.json()

File: lib/services.ts
// This file is for reference only - showing how your Django backend works
// It's not used directly in the frontend code

export async function generateApiKey(token: string) {
  const BASE_URL = "https://apis.omeife.ai/api/v1/"
  const url = `${BASE_URL}user/developer/generate-key`
  const headers = {
    Authorization: `Bearer ${token}`,
    "Content-Type": "application/json",
  }

  try {
    const response = await fetch(url, { headers })
    if (!response.ok) {
      throw new Error(`Failed to generate API key: ${response.status}`)
    }
    const data = await response.json()
    return data?.data?.key
  } catch (error) {
    console.error("Error generating API key:", error)
    return null
  }
}

export async function makeRequest(
  endpoint: string,
  data: any,
  apiKey: string | null = null,
  token: string | null = null,
) {
  if (!apiKey && token) {
    apiKey = await generateApiKey(token)
  }

  if (!apiKey) {
    return { error: "Failed to retrieve API key" }
  }

  const BASE_URL = "https://apis.omeife.ai/api/v1/"
  const headers = {
    Authorization: `Bearer ${apiKey}`,
    "Content-Type": "application/vnd.api+json",
    Accept: "application/vnd.api+json",
  }

  const fullEndpoint = `${BASE_URL}${endpoint}`

  try {
    const response = await fetch(fullEndpoint, {
      method: "POST",
      headers,
      body: JSON.stringify(data),
    })

    if (!response.ok) {
      throw new Error(`API request failed: ${response.status} ${response.statusText}`)
    }

    return await response.json()
  } catch (error) {
    console.error("API request failed:", error)
    return { error: "Request failed", details: String(error) }
  }
}

